discovery.md

```diff
@@ -109,296 +123,218 @@ to execute some logic, which might lead to the occurrence of new events.
-discovery system.
-
-##### `service` entity attributes
+##### name
+- Type: `String`
+- Description: A unique identifier for this Service. This value MUST be
+  unique within the scope of this Discovery Endpoint.
+- Constraints:
+  - REQUIRED
+  - MUST be a valid `fsegment` per RFC1738.
+- Examples:
+  - `storage`
+  - `github`
 
-###### name
+##### url
 
-- Type: `String`
-- Description: The `name` attribute SHOULD be human readable and can identify a
-  top level service.
+- Type: `URL`
+- Description: A URL that references this Service. This value MUST
+  be usable in subsequent requests, by authorized clients, to retrieve this
+  Service entity.
 - Constraints:
   - REQUIRED
-  - MUST be a non-empty string
+  - MUST be a non-empty URL
+  - MUST end with `fsegments` (per RFC1738) of: `/services/{name}` where
+    `name` is the Service's `name` attribute.
 - Examples:
-  - "GitHub"
-  - "Awesome Cloud Storage"
-  - "com.example.microservices.userlogin"
+  - `http://example.com/services/storage`
+  - `http://discovery.github.com/services/github`
 
-###### description
+##### description (Service)
 
 - Type: `String`
-- Description: Human readable description .
+- Description: Human readable description.
 - Constraints:
   - OPTIONAL
   - If present, MUST be a non-empty string
 
-###### service_uri
+##### docsurl
 
-- Type: `URI`
-- Description: Absolute URI that provides a link back to the producer, or
-  documentation about the producer. This is intended for a developer to
-  use in order to learn more about this service events produced.
+- Type: `URL`
+- Description: Absolute URL that provides a link to additional documentation
+  about the service.  This is intended for a developer to
+  use in order to learn more about this service's events produced.
 - Constraints:
   - OPTIONAL
   - If present, MUST be a non-empty absolute URI
 - Examples:
-  - "http://cloud.example.com/docs/blobstorage"
-
-
-##### `service` entity examples
-
-```json
-{
-  "name": "Cloud Storage Service",
-  "description": "Blob storage in the cloud",
-  "service_uri": "https://cloud.example.com/docs/storage"
-}
-```
-
-And a list of valid `service` entities.
-
-```json
-[
-  {
-    "name": "Cloud Storage Service",
-    "description": "Blob storage in the cloud",
-    "service_uri": "https://cloud.example.com/docs/storage"
-  },
-  {
-    "name": "Cloud MySQL"
-  },
-  {
-    "name": "Cloud OtherSQL",
-    "description": "Highly scalable SQL service"
-  }
-]
-```
-
-#### `type` entity
+  - `http://cloud.example.com/docs/blobstorage`
 
-Used in discovery for enumerating the different CloudEvent `type`s represented
-in this discovery system.
-
-##### `type` entity attributes
-
-###### type
+###### specversion
 
-- Type: `String`
-- Description: CloudEvents [`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type)
-  attribute.
+- Type: `String` per [RFC 2046](https://tools.ietf.org/html/rfc2046)
```

Comment:
I wonder, if this could also be an array, if multiple specversions are supported, but that would also require the subscription API to support subscribing for an event in a specific specversion.

Reply:
For some reason this makes me nervous, but I can't explain why.  It does make sense though so I'll make this `specversions` and see what people think.